content/en/blog/_posts/2023-12-05-kubernetes-1.29.md

```diff
@@ -0,0 +1,183 @@
+
+### Implement In-place Update of Pod Resources ([sig-windows](https://github.com/kubernetes/community/tree/master/sig-windows))
+
+## Graduations, deprecations and removals for Kubernetes 1.29
+
+### Graduated to stable
+
+This release includes a total of 11 enhancements promoted to Stable:
+
+- [Remove transient node predicates from KCCM's service controller](https://github.com/kubernetes/enhancements/issues/3458)
+- [Reserve nodeport ranges for dynamic and static allocation](https://github.com/kubernetes/enhancements/issues/3668)
+- [Priority and Fairness for API Server Requests](https://github.com/kubernetes/enhancements/issues/1040)
+- [KMS v2 Improvements](https://github.com/kubernetes/enhancements/issues/3299)
+- [Support paged LIST queries from the Kubernetes API](https://github.com/kubernetes/enhancements/issues/365)
+- [ReadWriteOncePod PersistentVolume Access Mode](https://github.com/kubernetes/enhancements/issues/2485)
+- [Kubernetes Component Health SLIs](https://github.com/kubernetes/enhancements/issues/3466)
+- [CRD Validation Expression Language](https://github.com/kubernetes/enhancements/issues/2876)
+- [Introduce nodeExpandSecret in CSI PV source](https://github.com/kubernetes/enhancements/issues/3107)
+- [Track Ready Pods in Job status](https://github.com/kubernetes/enhancements/issues/2879)
+- [Kubelet Resource Metrics Endpoint](https://github.com/kubernetes/enhancements/issues/727)
+
+### Deprecations and removals
+
+#### Removal of in-tree integrations with cloud providers ([sig-cloud-provider](https://github.com/kubernetes/community/tree/master/sig-cloud-provider))
+
+Kubernetes 1.29 defaults to operating _without_ a built-in integration to any cloud provider.
+If you have previously been relying on in-tree cloud provider integrations (with Azure, GCE, or vSphere) then you can either:
+- enable an equivalent external [cloud controller manager](/docs/concepts/architecture/cloud-controller/)
+  integration _(recommended)_
+- opt back in to the legacy integration by setting the associated feature gates to `false`; the feature
+  gates to change are `DisableCloudProviders` and `DisableKubeletCloudCredentialProviders`
+  
+Enabling external cloud controller managers means you must run a suitable cloud controller manager within your cluster's control plane; it also requires setting the command line argument `--cloud-provider=external` for the kubelet (on every relevant node), and across the control plane (kube-apiserver and kube-controller-manager).
+
+For more information about how to enable and run external cloud controller managers, read [Cloud Controller Manager Administration](/docs/tasks/administer-cluster/running-cloud-controller/) and [Migrate Replicated Control Plane To Use Cloud Controller Manager](/docs/tasks/administer-cluster/controller-manager-leader-migration/).
+
```

Comment:
we talked about this entry during the sig cloud provider meeting today and it was suggested that we add the links to the cloud controller managers here as well. could we add the follow:

```
If you need a cloud controller manager for one of the legacy in-tree providers, please see the following links:

* AWS - https://github.com/kubernetes/cloud-provider-aws 
* Azure - https://github.com/kubernetes-sigs/cloud-provider-azure
* GCE - https://github.com/kubernetes/cloud-provider-gcp
* OpenStack - https://github.com/kubernetes/cloud-provider-openstack
* vSphere - https://github.com/kubernetes/cloud-provider-vsphere 
```